File: static/csharp/FormulaId.cs
namespace MyAirtable;

/// <summary>Formula builder for record IDs.</summary>
public sealed class FormulaId
{
    /// <summary>Match a specific record ID.</summary>
    public string Eq(string id) => "RECORD_ID()='" + Formulas.EscapeFormulaStringSingle(id) + "'";

    /// <summary>Match any of the given record IDs.</summary>
    public string InList(IReadOnlyList<string> ids) =>
        ids.Count switch
        {
            0 => "FALSE()",
            1 => Eq(ids[0]),
            _ => Formulas.Or(ids.Select(Eq)),
        };
}

File: src/myairtable/static/java/FormulaId.java
// ==========================================
// MyAirtable static runtime.
// ==========================================

package myairtable;

import java.util.List;

/** Formula builder for record IDs. */
public final class FormulaId {

  /** Match a specific record ID. */
  public String eq(String id) {
    return "RECORD_ID()='" + Formulas.escapeFormulaStringSingle(id) + "'";
  }

  /** Match any of the given record IDs. */
  public String inList(List<String> ids) {
    return switch (ids.size()) {
      case 0 -> "FALSE()";
      case 1 -> eq(ids.get(0));
      default -> Formulas.or(ids.stream().map(this::eq).toList());
    };
  }
}
